src/common/io/socket/common.c

```diff
@@ -129,6 +141,52 @@ sckOptionSet(int fd)
     FUNCTION_TEST_RETURN_VOID();
 }
 
+/**********************************************************************************************************************************/
+static bool
+sckConnectInProgress(int errNo)
+{
+    return errNo == EINPROGRESS || errNo == EINTR;
+}
+
+void
+sckConnect(int fd, const String *host, unsigned int port, const struct addrinfo *hostAddress, TimeMSec timeout)
+{
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(INT, fd);
+        FUNCTION_LOG_PARAM(STRING, host);
+        FUNCTION_LOG_PARAM(UINT, port);
+        FUNCTION_LOG_PARAM_P(VOID, hostAddress);
+        FUNCTION_LOG_PARAM(TIME_MSEC, timeout);
+    FUNCTION_LOG_END();
+
+    // Attempt connection
+    if (connect(fd, hostAddress->ai_addr, hostAddress->ai_addrlen) == -1)
+    {
+        // Save the error
+        int errNo = errno;
+
+        // The connection has started but since we are in non-blocking mode it has not completed yet
+        if (sckConnectInProgress(errNo))
+        {
+            // Wait for write-ready
+            if (!sckReadyWrite(fd, timeout))
```

Comment:
I'm guessing choosing to sckReadyWrite vs sckReadyRead was arbitrary?

Reply:
A socket won't be read ready until the other side writes something, but it will be write ready as soon as the connection is complete.